admin.go

```diff
@@ -136,11 +136,12 @@ func Forever() int64 {
 }
 
 // Ban will ban user from chat until `member.RestrictedUntil`.
-func (b *Bot) Ban(chat *Chat, member *ChatMember) error {
+func (b *Bot) Ban(chat *Chat, member *ChatMember, revokeMessages bool) error {
```

Comment:
We can't just add another argument to the widely used `Ban` function, because it will break backward compatibility. And since `revoke_messages` doesn't fit in `ChatMember` structure, the only option is to make `revokeMessages` variadic and if the additional boolean argument was given, to check the length and reassign the local `revoke` variable, before passing it to the map.

Reply:
Even though the field should not be added to the ChatMember structure, the accepted commit still does it https://github.com/tucnak/telebot/pull/407/commits/50ae1bb6b2bb4925affbd09dee8450cce3ea814b#diff-370ab673a241352de628b0935cae